src/fluree/db/ledger/upgrade/full_text.clj

```diff
@@ -0,0 +1,120 @@
+(ns fluree.db.ledger.upgrade.full-text
+  (:require [fluree.db.api :as fdb]
+            [fluree.db.constants :as const]
+            [fluree.db.dbproto :as dbproto]
+            [fluree.db.server :as server]
+            [fluree.db.query.range :as query-range]
+            [clojure.core.async :as async :refer [<!!]]
+            [clojure.tools.logging :as log]
+            [environ.core :as environ])
+  (:import (fluree.db.flake Flake)))
+
+(set! *warn-on-reflection* true)
+
+(def pred-name "_predicate/fullText")
+(def invalid-name "_predicate/fullText-invalid")
+
+(defn wait-for-block
+  [conn ledger expected-block]
+  (loop [{:keys [block] :as db} (<!! (fdb/db conn ledger))]
```

Comment:
@bplatz looking for other options than a Thread/sleep loop.  

- would sync-to be an option? We have the expected block.
- or would it better to use monitor-tx (with a very long expiration) for the last submitted transaction?

Reply:
You can use: `(<!! (fdb/db conn ledger {:syncTo expected-block}))`